tests/signature/test_parsing.py

```diff
@@ -284,3 +289,200 @@ def handler(param: bool) -> None:
         response = client.get(f"/?param={query}")
         assert response.status_code == HTTP_200_OK, response.json()
         mock.assert_called_once_with(expected)
+
+
+@pytest.mark.parametrize("preferred_validation_backend", ("attrs", "pydantic"))
+def test_validation_error_exception_key(preferred_validation_backend: Literal["attrs", "pydantic"]) -> None:
+    class OtherChild(BaseModel):
+        val: List[int]
+
+    class Child(BaseModel):
+        val: int
+        other_val: int
+
+    class Parent(BaseModel):
+        child: Child
+        other_child: OtherChild
+
+    def fn(model: Parent) -> None:
+        pass
+
+    model = create_signature_model(
+        fn=fn,
+        dependency_name_set=set(),
+        preferred_validation_backend=preferred_validation_backend,
+        parsed_signature=ParsedSignature.from_fn(fn, {}),
+    )
+
+    with pytest.raises(ValidationException) as exc_info:
+        model.parse_values_from_connection_kwargs(
+            connection=RequestFactory().get(), model={"child": {}, "other_child": {}}
+        )
+
+    assert isinstance(exc_info.value.extra, list)
+    assert exc_info.value.extra[0]["key"] == "model.child.val"
+    assert exc_info.value.extra[1]["key"] == "model.child.other_val"
+    assert exc_info.value.extra[2]["key"] == "model.other_child.val"
+
+
+def test_invalid_input_pydantic() -> None:
+    class OtherChild(BaseModel):
+        val: List[int]
+
+    class Child(BaseModel):
+        val: int
+        other_val: int
+
+    class Parent(BaseModel):
+        child: Child
+        other_child: OtherChild
+
+    @post("/")
+    def test(
+        data: Parent,
+        int_param: int,
+        int_header: int = Parameter(header="X-SOME-INT"),
+        int_cookie: int = Parameter(cookie="int-cookie"),
+    ) -> None:
+        ...
+
+    with create_test_client(route_handlers=[test]) as client:
+        response = client.post(
+            "/",
+            json={"child": {"val": "asdf", "other_val": "asdf"}, "other_child": {"val": [1, "asdf"]}},
+            params={"int_param": "asdf"},
+            headers={"X-SOME-INT": "asdf"},
+            cookies={"int-cookie": "asdf"},
+        )
+
+        assert response.status_code == HTTP_400_BAD_REQUEST
+
+        data = response.json()
+
+        assert data
+        assert data["extra"] == [
+            {"key": "child.val", "message": "value is not a valid integer", "source": "body"},
+            {"key": "child.other_val", "message": "value is not a valid integer", "source": "body"},
+            {"key": "other_child.val.1", "message": "value is not a valid integer", "source": "body"},
```

Comment:
This isn't tested against attrs, does it behave the same?

Reply:
Good catch. Attrs actually fails this check when there are multiple nested models. Working on a fix now.

Reply:
Fixed, @peterschutt !